test/IntegrationTests/Helpers/TestHttpListener.cs

```diff
@@ -79,11 +79,37 @@ public TestHttpListener(ITestOutputHelper output, Action<HttpListenerContext> re
     /// </summary>
     public int Port { get; }
 
-    public Task<bool> VerifyHealthzAsync()
+    public Action<HttpListenerContext> Handler
     {
-        var healhtzEndpoint = $"{_prefix.Replace("*", "localhost")}/healthz";
+        get
+        {
+            lock (_requestHandlerLocker)
+            {
+                return _requestHandler;
+            }
+        }
 
-        return HealthzHelper.TestHealtzAsync(healhtzEndpoint, nameof(MockLogsCollector), _output);
+        set
+        {
+            lock (_requestHandlerLocker)
+            {
+                _requestHandler = value;
+            }
+        }
+    }
```

Comment:
This can be improved (even to avoid locking).
However, I do not want to make the code harder to read (I do not think it would ever cause performance issues).